stock_operation_type_location/wizard/stock_transfer_details.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# © 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, models
+from lxml import etree
+
+
+class StockTransferDetails(models.TransientModel):
+    _inherit = "stock.transfer_details"
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(StockTransferDetails, self)\
+            .fields_view_get(
+                view_id=view_id, view_type=view_type,
+                toolbar=toolbar, submenu=submenu)
+        context = self._context
+        loc = []
+        dest_loc = []
+        domain_sourceloc = ""
+        domain_destinationloc = ""
+        obj_stock_picking_type = self.env['stock.picking.type']
+
+        picking_type_id = context.get('default_picking_type_id')
+
+        if picking_type_id:
+            picking_type = obj_stock_picking_type.browse(
+                [picking_type_id])[0]
+
+            allowed_location = picking_type.allowed_location_ids
+            allowed_dest_location = picking_type.allowed_dest_location_ids
+            scr_id = picking_type.default_location_src_id.id
+            dest_id = picking_type.default_location_dest_id.id
+
+            if allowed_location:
+                for location in allowed_location:
+                    loc.append(location.id)
+                if scr_id and scr_id not in loc:
+                    loc.append(scr_id)
+
+            if allowed_dest_location:
+                for dest_location in allowed_dest_location:
+                    dest_loc.append(dest_location.id)
+                if dest_id and dest_id not in dest_loc:
+                    dest_loc.append(dest_id)
+
+            if loc and dest_loc:
+                domain_sourceloc = "[('id', 'in', %s)]" % loc
+                domain_destinationloc = "[('id', 'in', %s)]" % dest_loc
+            elif loc and not dest_loc:
+                domain_sourceloc = "[('id', 'in', %s)]" % loc
+                domain_destinationloc = "[]"
+            elif not loc and dest_loc:
+                domain_sourceloc = "[]"
+                domain_destinationloc = "[('id', 'in', %s)]" % dest_loc
+            else:
+                domain_sourceloc = "[]"
+                domain_destinationloc = "[]"
```

Comment:
you could much simplify this block by first initializaing domain_sourceloc and domain_destinationloc to "[]" and having two blocks changing the value `if loc:` and `if dest_loc:`